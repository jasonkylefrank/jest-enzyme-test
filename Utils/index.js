
// Helper for Jest find() calls which target a data-testid attribute
export const findByTestIdAttr = (component, attr) => {
    return component.find(`[data-testid="${attr}"]`);
};