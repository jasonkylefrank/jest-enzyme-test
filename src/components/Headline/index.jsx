import React, { Component } from 'react';

class Headline extends Component {
    // constructor(props) {
    //     super(props);
    // }

    render() {
        const { header, desc } = this.props;

        if (!header) {
            return null;
        }

        return (
            <div data-testid="Headline-root">
                <h1 data-testid="header">{header}</h1>
                <p data-testid="desc">{desc}</p>
            </div>
        );
    }
}

export default Headline;
