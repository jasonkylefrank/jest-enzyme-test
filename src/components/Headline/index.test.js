import React from 'react';
import { shallow } from 'enzyme';
import Headline from './index';
import { findByTestIdAttr } from '../../../Utils';

describe('Headline component', () => {
    const setUp = (props = {}) => (shallow(<Headline {...props} />));

    describe('Have props', () => {
        let component;

        beforeEach(() => {
            const props = {
                header: 'Test Header',
                desc: 'Test description'
            };
            component = setUp(props);
        });

        it('Should render without errors', () => {
            const element = findByTestIdAttr(component, "Headline-root");
            expect(element.length).toBe(1);
        });

        it('Should render an H1', () => {
            const element = findByTestIdAttr(component, "header");
            expect(element.length).toBe(1);
        });

        it('Should render a description', () => {
            const element = findByTestIdAttr(component, "desc");
            expect(element.length).toBe(1);
        });
    });
    
    describe('Have NO props', () => {
        let component;
        
        beforeEach(() => {            
            component = setUp();
        });

        it('Should not render', () => {
            const element = findByTestIdAttr(component, "Headline-root");
            expect(element.length).toBe(0);
        })
    });
    
});
