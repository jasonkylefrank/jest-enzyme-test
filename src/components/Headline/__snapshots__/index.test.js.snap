// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component Have props Matches the snapshot 1`] = `
<div
  data-testid="Headline-root"
>
  <h1
    data-testid="header"
  >
    Test Header
  </h1>
  <p
    data-testid="desc"
  >
    Test description
  </p>
</div>
`;
